internal/objects/validator.go

```diff
@@ -347,7 +376,10 @@ func (v *Validator) hasConflict(inst *unstructured.Unstructured) (bool, []string
 			// If the user have chosen not to propagate the object to this descendant,
 			// there shouldn't be any conflict reported here
 			nsLabels := v.Forest.Get(inst.GetNamespace()).GetLabels()
-			if ok, _ := selectors.ShouldPropagate(inst, nsLabels); ok {
+			fmt.Println("fuck1")
+			propogateByDefault := v.isPropagateType(metav1.GroupVersionKind(gvk))
+			fmt.Println("fuck2")
+			if ok, _ := selectors.ShouldPropagate(inst, nsLabels, propogateByDefault); ok {
```

Comment:
Maybe just allow selectors.ShouldPropagate take the mode, so you can encapsulate the logic in there? It'll make the calling code shorter and nicer IMO.